### PR TITLE
sugest to avoid to memory leak

### PR DESCRIPTION
**Pull request by MatejMagat305**
avoid memory leak resume and sugest to avoid to call GO functions for empty life cycle functions ...

### CODE REVIEW

File: internal/driver/mobile/app/android.c [hunk 1]
```diff
@@ -70,10 +70,12 @@ static int main_running = 0;
 void processOnResume(ANativeActivity *activity) {
 	JNIEnv* env = activity->env;
 	setCurrentContext(activity->vm, (*env)->NewGlobalRef(env, activity->clazz));
-
-    onResume(activity);
```
Comment (andydotxyz) [inline internal/driver/mobile/app/android.c:74]: This cannot be right can it?
It looks like you have moved 3 of the callbacks from Go to C but deleted this one?

**Review by andydotxyz — COMMENTED** [review]
I'm not sure about the way that some callbacks are deleted and others moved.

Why not make the PR just about the memory leak fix?

**Comment by MatejMagat305** [comment]
> I'm not sure about the way that some callbacks are deleted and others moved.
> 
> Why not make the PR just about the memory leak fix?

this lines: https://github.com/fyne-io/fyne/pull/4118/files#diff-86b6487000ebad8b8c46272d0aa615cdbec1dba15013d82bed007ffa593e5464R90 is memory leaks, others are only therefore nothing theese functions do and call GO from C is recomended to eliminate ...

**Comment by Jacalz** [comment]
But you have, as stated in https://github.com/fyne-io/fyne/pull/4118#discussion_r1289639647,  removed one of the functions. Why?

**Comment by andydotxyz** [comment]
I think I see why it looks inconsistent - because the `onResume` was written in C to do something but not in Go. But it was called `processOnResume`. If that was renamed to `onResume` in the C file then I think we would have a consistent code and this could be approved.
Is that OK @MatejMagat305 ?

**Comment by MatejMagat305** [comment]
ok

**Comment by Jacalz** [comment]
Why did you close it? Andy just suggested changes

**Comment by MatejMagat305** [comment]
ach soory, I will changes tomorow

**Review by andydotxyz — APPROVED** [review]
Thanks for making these optimisations

**Review by Jacalz — APPROVED** [review]
Thanks. If there are more places where we are passing Go functions back into C, we probably want to remove those as well some time in the future. It is quite slow from what I have understood (but a lot faster in Go 1.21).

**Comment by MatejMagat305** [comment]
> Thanks. If there are more places where we are passing Go functions back into C, we probably want to remove those as well some time in the future. It is quite slow from what I have understood (but a lot faster in Go 1.21).

well there are some functions like `onContentRectChanged` or `onNativeWindowCreated`, which are empty, but I did not know which one will you plan use ..., life time I dare move because I saw that life time are binding with running loop (are call from other way ...)

**Pull request merged**